modules/dnn/src/onnx/onnx_importer.cpp

```diff
@@ -3183,6 +2851,44 @@ void ONNXImporter::parseCumSum(LayerParams& layerParams, const opencv_onnx::Node
     addLayer(layerParams, node_proto);
 }
 
+// "Equal" "Greater" "Less" "Pow" "Add" "Sub" "Mul" "Div" "Sum" "Min" "Max"
+void ONNXImporter::parseElementWise(LayerParams& layerParams, const opencv_onnx::NodeProto& node_proto)
```

Comment:
I think we might need to add some kind of a workaround so that we know original dimensions of data, otherwise tests fail because we treated shape (2,) as (2,1) instead of (1,2), for example.

Reply:
An extra variable is added to carry the original dimensions of constants. Once the original dimension is 1 and the input constant has two dimensions and the last dimension of 1, for example (2, 1), then the input constant will be transposed to (1, 2).

I would not say this solution is elegant and robust enough, but it may be a suitable one until we have 1-d Mat.

Reply:
I reuse some code from `parseMul` to handle cases of two non-constant inputs, which should help us to pass all tests without disabling any.  Once one-dimensional Mat is supported (https://github.com/opencv/opencv/pull/18594), code related to blobs of extra info and the workaround of two non-constant inputs should be deleted.
 https://github.com/opencv/opencv/blob/b91f173680af92498a8bd2517fc7c524c2eae640/modules/dnn/src/onnx/onnx_importer.cpp#L2290-L2351

Reply:
We'll need to know the exact shapes for Gather layer anyways, if it's possible to do with extra info, but you think it's unjustified. I'm just sad that we'll still use other layers that don't support broadcasting.

Reply:
In the case of two non-constant input but one of them is one-dimensional, when net.forward() is called, the first time `getMemoryShape()` is called it can get the correct shape, but the second time `getMemoryShape()` is called it will get the wrong shape since we load a one-dimensional tensor of shape [2] as a Mat of shape [2, 1]. This will lead to an error in `findCommonShape()`. In this case, I do not think of a workaround as what I did for `parseElementWise()` because that shape is got in the forward stage.

Reply:
@rogday There are two such cases in ONNX conformance: `Test_ONNX_conformance.Layer_Test/test_div_bcast_OCV_CPU` and `Test_ONNX_conformance.Layer_Test/test_mul_bcast_OCV_CPU`.

To be honest, we can disable these two tests and turn them on when we have 1d Mat support. Current implementation without custom div and mul layers can pass all other tests of layers and nets (check [this log](https://github.com/opencv/opencv/actions/runs/2589842789)). Pratically speaking, I would call the two tests rare cases, since internels being 1d tensor are rare for conv nets.

Reply:
I agree.